zerver/models.py

```diff
@@ -320,11 +332,20 @@ def get_realm_by_email_domain(email):
     if settings.REALMS_HAVE_SUBDOMAINS:
         raise GetRealmByDomainException(
             "Cannot get realm from email domain when settings.REALMS_HAVE_SUBDOMAINS = True")
-    try:
-        alias = RealmAlias.objects.select_related('realm').get(domain = email_to_domain(email))
+    domain = email_to_domain(email)
+    query = RealmAlias.objects.select_related('realm')
+    alias = query.filter(domain=domain, subdomains_allowed=False).first() or \
```

Comment:
> You could add the first filter into the query, so query = RealmAl...().filter(domain=domain) in the line above.
It can't be moved into the query since just below this line of code I use the same query object for matching the subdomains.
```
   domain = email_to_domain(email)
    query = RealmAlias.objects.select_related('realm')
    alias = query.filter(domain=domain, subdomains_allowed=False).first() or \
        query.filter(domain=domain, subdomains_allowed=True).first()
    if alias:
        return alias.realm
    else:
        query = query.filter(subdomains_allowed=True)
        while domain:
            subdomain, sep, domain = domain.partition('.')
            alias = query.filter(domain=domain).first()
            if alias:
                return alias.realm
```